src/App/Datums.h

```diff
@@ -0,0 +1,203 @@
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef DATUMS_H
+#define DATUMS_H
+
+#include "GeoFeature.h"
+
+#include "GeoFeatureGroupExtension.h"
+
+#include "QCoreApplication"
+
+namespace App
+{
+    class LocalCoordinateSystem;
+
+class AppExport DatumElement : public App::GeoFeature
+{
+    PROPERTY_HEADER_WITH_OVERRIDE(App::DatumElement);
+public:
+    /// additional information about the feature usage (e.g. "BasePlane-XY" or "Axis-X")
+    PropertyString Role;
+
+    /// Constructor
+    DatumElement(bool hideRole = true);
+    ~DatumElement() override;
+
+    /// Finds the origin object this plane belongs to
+    App::LocalCoordinateSystem* getLCS();
+
+    /// Returns true if this DatumElement is part of a App::Origin.
+    bool isOriginFeature();
+};
+
+class AppExport Plane: public App::DatumElement {
+    PROPERTY_HEADER_WITH_OVERRIDE(App::DatumElement);
+public:
+    const char* getViewProviderName() const override {
+        return "Gui::ViewProviderPlane";
+    }
+};
+
+class AppExport Line: public App::DatumElement {
+    PROPERTY_HEADER_WITH_OVERRIDE(App::DatumElement);
+public:
+    const char* getViewProviderName() const override {
+        return "Gui::ViewProviderLine";
+    }
+};
+
+class AppExport Point: public App::DatumElement {
+    PROPERTY_HEADER_WITH_OVERRIDE(App::DatumElement);
+public:
+    const char* getViewProviderName() const override {
+        return "Gui::ViewProviderPoint";
+    }
+};
+
+
+class AppExport LocalCoordinateSystem : public App::GeoFeature
+{
+    PROPERTY_HEADER_WITH_OVERRIDE(App::LocalCoordinateSystem);
+    Q_DECLARE_TR_FUNCTIONS(App::LocalCoordinateSystem)
+
+public:
+    /// Constructor
+    LocalCoordinateSystem();
+    ~LocalCoordinateSystem() override;
+
+    /// returns the type name of the ViewProvider
+    const char* getViewProviderName() const override {
+        return "Gui::ViewProviderOrigin";
```

Comment:
Why not rename view provider to `ViewProviderLocalCoordinateSystem`?

Reply:
I wanted to limit the size of the PR because it is already so big. And renaming files just touch everywhere. Do you think I should still do it now?

Reply:
Imho yes because if not now you probably will never do it and in few years we all will wonder why such inconsistency exists.

Reply:
Sounds good to me.